tests/integration/test_main.py

```diff
@@ -94,9 +133,71 @@ def test_empty_run(tmp_path, monkeypatch):
     config_file = config_dir / "config.yml"
     config_file.write_text(f"output_dir: {log_dir}")
 
-    with pytest.raises(RecipeError) as exc:
-        ESMValTool().run(recipe_file, config_dir=config_dir)
-    assert str(exc.value) == "The given recipe does not have any diagnostic."
+    err_msg = "The given recipe does not have any diagnostic."
+    warn_msg = "Please use the option `config_dir` instead"
+    with pytest.raises(RecipeError, match=err_msg):
+        with pytest.warns(ESMValCoreDeprecationWarning, match=warn_msg):
+            ESMValTool().run(recipe_file, config_file=config_file)
+    log_file = os.path.join(
+        log_dir,
+        os.listdir(log_dir)[0],
+        "run",
+        "main_log.txt",
+    )
+    filled_recipe = os.path.exists(
+        log_dir + "/" + os.listdir(log_dir)[0] + "/run/recipe_filled.yml",
```

Comment:
Would it be worth using `os.path.join` here too?

Reply:
Good point! I think the logic in that test was also flawed (it called `shutil.rmtree` and then asserts that a file does not exist within the removed dir?). Fixed that in https://github.com/ESMValGroup/ESMValCore/pull/2736/commits/078b8b303a507397afb46f90d212d8665529520f